dll/win32/browseui/CAutoComplete.cpp

```diff
@@ -25,664 +25,2124 @@
-                                    TRACE("HeapAlloc failed to allocate %d bytes\n", (len + 1) * sizeof(WCHAR));
-                                }
-                            }
-                            else
-                            {
-                                SendMessageW(hwnd, WM_SETTEXT, 0, (LPARAM)pThis->m_txtbackup);
-                                SendMessageW(hwnd, EM_SETSEL, wcslen(pThis->m_txtbackup), wcslen(pThis->m_txtbackup));
-                            }
-                        }
-                        return 0;
-                    }
-                }; break;
-                
-                case VK_BACK:
-                {
-                    if (GetKeyState(VK_CONTROL) < 0) // Ctrl+Backspace
-                    {
-                        Edit_BackWord(hwnd);
-                        return 0;
-                    }
-                }
-                // FALL THROUGH
-                case VK_DELETE:
-                {
-                    if ((! *hwndText) && (pThis->m_options & ACO_AUTOSUGGEST))
-                    {
-                        ShowWindow(pThis->m_hwndListBox, SW_HIDE);
-                        return CallWindowProcW(pThis->m_wpOrigEditProc, hwnd, uMsg, wParam, lParam);
-                    }
-                    
-                    if (pThis->m_options & ACO_AUTOAPPEND)
-                    {
-                        DWORD b;
-                        SendMessageW(hwnd, EM_GETSEL, (WPARAM)&b, (LPARAM)NULL);
-                        if (b>1)
-                        {
-                            hwndText[b-1] = '\0';
-                        }
-                        else
-                        {
-                            hwndText[0] = '\0';
-                            SetWindowTextW(hwnd, hwndText);
-                        }
-                    }
-                }; break;
-                
-                default:
-                    ;
-            }
+// WM_NCHITTEST
+LRESULT CACListView::OnNCHitTest(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    TRACE("CACListView::OnNCHitTest(%p)\n", this);
+    ATLASSERT(m_pDropDown);
+    POINT pt = { GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam) }; // in screen coordinates
+    ScreenToClient(&pt); // into client coordinates
+    HWND hwndTarget = m_pDropDown->ChildWindowFromPoint(pt);
+    if (hwndTarget != m_hWnd)
+        return HTTRANSPARENT; // pass through (for resizing the drop-down window)
+    bHandled = FALSE; // do default
+    return 0;
+}
 
-            SendMessageW(pThis->m_hwndListBox, LB_RESETCONTENT, 0, 0);
+// WM_RBUTTONDOWN @implemented
+LRESULT CACListView::OnRButtonDown(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    TRACE("CACListView::OnRButtonDown(%p)\n", this);
+    return 0; // eat
+}
 
-            HeapFree(GetProcessHeap(), 0, pThis->m_txtbackup);
+// WM_RBUTTONUP @implemented
+LRESULT CACListView::OnRButtonUp(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
```

Comment:
The same comment as for the middle button may apply as well for the right button.

Reply:
United.